main/init/src/mill/init/maven/codegen.scala

```diff
@@ -0,0 +1,353 @@
+            else
+              s"""
+                 |${comment("required to access base package in root build file")}
+                 |import $$file.$baseModule""".stripMargin
+          }
+           |
+           |object `package` extends RootModule with $baseModule""".stripMargin
+
+      val source = model.getPackaging match {
+        case "pom" =>
+          s"""$header {
+             |
+             |  override def artifactName = "${model.getArtifactId}"
+             |
+             |  override def pomPackagingType = PackagingType.Pom
+             |}
+             |""".stripMargin
+
+        // this should be limited to jar but having some scaffolding might be useful
+        case _ =>
+          val javacOptions = plugin.`maven-compiler-plugin`.javacOptions(model)
+          val (moduleDeps, ivyDeps, compileIvyDeps, runIvyDeps, testIvyDeps) = deps(model, modules)
+          val testSource =
+            if (testIvyDeps.isEmpty) ""
+            else
+              s"""${
+                  if (testModule.isEmpty)
+                    s"""  ${comment("\"testFramework\" needs to be specified")}
+                       |  trait Tests extends super.Tests""".stripMargin
+                  else "  object test extends Tests"
+                } {
+                 |
+                 |    override def ivyDeps = super.ivyDeps() ++ ${
+                  break.cs.indent(3, "Agg(", testIvyDeps, ")")
+                }
+                 |  }
+                 |""".stripMargin
+
+          s"""$header {
+             |
+             |  override def artifactName = "${model.getArtifactId}"
+             |${
+              if (javacOptions.isEmpty) ""
+              else
+                s"""
+                   |  override def javacOptions = ${
+                    break.cs.indent(2, "Seq(", javacOptions.iterator.map(escape), ")")
+                  }
+                   |""".stripMargin
+            }${
+              if (moduleDeps.isEmpty) ""
+              else
+                s"""
+                   |  override def moduleDeps = ${break.cs.indent(2, "Seq(", moduleDeps, ")")}
+                   |""".stripMargin
+            }${
+              if (ivyDeps.isEmpty) ""
+              else
+                s"""
+                   |  override def ivyDeps = ${break.cs.indent(2, "Agg(", ivyDeps, ")")}
+                   |""".stripMargin
+            }${
+              if (compileIvyDeps.isEmpty) ""
+              else
+                s"""
+                   |  override def compileIvyDeps = ${
+                    break.cs.indent(2, "Agg(", compileIvyDeps, ")")
+                  }
+                   |""".stripMargin
+            }${
+              if (runIvyDeps.isEmpty) ""
+              else
+                s"""
+                   |  override def runIvyDeps = ${break.cs.indent(2, "Agg(", runIvyDeps, ")")}
+                   |""".stripMargin
+            }
+             |
+             |$testSource
+             |}""".stripMargin
```

Comment:
There is a lot of code here related to managing indentation and layout, and it really gets in the way of understanding the core logic of what we are generating. 

Let's instead just generate the code naively - i.e. not worrying about managing indentation or proper number of blank lines - and just running `ScalafmtWorkerModule.worker().reformat()` on the generated files (we may need to bump the scalafmt version to the latest release so it can pick up `.mill` files)